esmvaltool/diag_scripts/shared/_base.py

```diff
@@ -517,9 +517,9 @@ def main(cfg):
     }
 
     output_directories = []
-    if cfg['write_netcdf']:
+    if cfg.get('write_netcdf', True):
         output_directories.append(cfg['work_dir'])
-    if cfg['write_plots']:
+    if cfg.get('write_plots', True):
```

Comment:
In my opinion we should not make the creation of these directories dependend on a custom diagnostic attribute. Suppose a user creates a diagnostic that writes `netcdf` and `csv` files to the `work` directory. Now when `write_netcdf` is set to `False`, the `work` directory is not created and the `csv` files cannot be written.

Reply:
Removed the code